### PR TITLE
Payments Work - Initial Followups

### LINKED ISSUE #545: Payments Followups
Top priority payments followups:

- Store next billing date separately from `expiresAt` to be able to present the right next billing date without grace periods causing any issues
- Add customer payment info to User model & remove from Subscription.
  - Merge `updateSubscriptionWithPaymentMethod` into `setPaymentMethod`
 - Mechanism for translating user-facing error messages that come from the server

### PR DESCRIPTION
**Pull request by robin-macpherson**
## Description

**Issue:** closes #545 

Handles the more essential payments followups that we identified when releasing Journaly Premium initially.

**Schema changes:** 
- Adds `nextBillingDate` to the `MembershipSubscription` model so that we can store the exact billing date independently of things like grace periods which are added behind the scenes and are not user-facing
- Moves a user's payment info-related fields off of the `MembershipSubscription` model and onto the `User` model

**Docs:**
- Some minor documentation updates related to the big Prisma upgrade we did in #570 , though these are not exhaustive

## Subtasks

- [x] I have added this PR to the Journaly Kanban project ✅
- [x] Test out setup using current docs to identify key changes
- [x] Make schema changes
- [x] Generate migs
- [x] Merge updateSubscriptionWithPaymentMethod into setPaymentMethod
- [x] Update all GQL queries that use the moved fields
- [x] Fix any build issues
- [x] Test all code & migs 

## Migrations

- CONTAINS migs to be applied before code is deployed **_AND backfill_**
- No downtime
- Manual SQL is required to backfill existing premium users' payment information
- These should be relatively low-risk since the payments flow is currently very rarely used. Let's just make sure to test really well on staging.

### CODE REVIEW

File: packages/j-db-client/\ [hunk 1]
```diff
@@ -0,0 +1,334 @@
+import {
```
Comment (robin-macpherson) [inline packages/j-db-client/\:1]: ... amazing 😓

File: packages/web/package.json [hunk 2]
```diff
@@ -41,7 +41,7 @@
   "dependencies": {
     "@apollo/client": "^3.3.12",
     "@apollo/react-ssr": "^4.0.0",
-    "@journaly/j-db-client": "^10.0.2",
+    "@journaly/j-db-client": "^11.0.0",
```
Comment (Lanny) [inline packages/web/package.json:44]: Oh, this reminds me, we also need to bump this version number in j-mail and remember to deploy the change

File: packages/web/scripts/grant-free-subscription.ts [hunk 3]
```diff
@@ -39,8 +39,6 @@ const handler = async (userId: number, months: number | undefined) => {
       expiresAt: expiresAt,
       stripeSubscription: stripeSubscription as unknown as Prisma.InputJsonValue,
       stripeSubscriptionId: stripeSubscription.id,
-      lastFourCardNumbers: 'NONE',
-      cardBrand: 'visa',
```
Comment (Lanny) [inline packages/web/scripts/grant-free-subscription.ts:43]: We should still apply this change to the user so we can present values (FE may choke if they have a subscription but these are null, not sure)

**Review by Lanny — CHANGES_REQUESTED** [review]
Looks good! Just blocking because the j-mail j-db-client dep upgrade, don't want to forget that and lose days of email again.

File: packages/web/package.json (hunk 2 above)
Comment (robin-macpherson) [reply packages/web/package.json:44]: Excellent catch!

File: packages/web/scripts/grant-free-subscription.ts (hunk 3 above)
Comment (robin-macpherson) [reply packages/web/scripts/grant-free-subscription.ts:43]: Great call 👍🏼

**Comment by robin-macpherson** [comment]
@Lanny great call blocking until those changes are made. The records in stage all updated correctly and as expected and I've now made the changes you highlighte and checked everything builds so will go ahead and override & merge 🚀

**Pull request merged**